engine/src/main/java/org/hibernate/search/engine/service/impl/StandardServiceManager.java

```diff
@@ -221,7 +221,7 @@ synchronized void stopVirtual() {
 				if ( status != ServiceStatus.RUNNING ) {
 					stateExpectedFailure();
 				}
-				stopAndRemoveFromCache();
+				stopReal();
```

Comment:
Yes, in fact that was the problem: one thread A would perform a stop operation and then remove the Service from the cache, while another thread B had already got a reference to it, yet had not started it. It will then start it (virtually), however by doing so it would increment the usage counter on a service wrapper which is no longer tracked by the registry.
Subsequent operations would initialize a new service wrapper, as the one in the registry was gone, and so the +1 and -1 operations end up being unbalanced.